roles/mysqld_exporter/meta/argument_specs.yml

```diff
@@ -0,0 +1,55 @@
+argument_specs:
+  main:
+    short_description: "Prometheus MySQLd Exporter"
+    description:
+      - "Deploy prometheus L(mysqld_exporter,https://github.com/prometheus/mysqld_exporter) using ansible"
+    author:
+      - "Prometheus Community"
+    options:
+      mysqld_exporter_version:
+        description: "MySQLd exporter package version. Also accepts latest as parameter."
+        default: "1.1.2"
+      mysqld_exporter_binary_local_dir:
+        description:
+          - "Enables the use of local packages instead of those distributed on github."
+          - "The parameter may be set to a directory where the C(mysqld_exporter) binary is stored on the host where ansible is run."
+          - "This overrides the I(mysqld_exporter_version) parameter"
+      mysqld_exporter_binary_url:
+        description: "URL of the mysqld_exporter binaries .tar.gz file"
+        default: "https://github.com/prometheus/mysqld_exporter/releases/download/v{{ mysqld_exporter_version }}/mysqld_exporter-{{ mysqld_exporter_version }}.linux-{{ go_arch }}.tar.gz"
+      mysqld_exporter_checksums_url:
+        description: "URL of the mysqld_exporter checksums file"
+        default: "https://github.com/prometheus/mysqld_exporter/releases/download/v{{ mysqld_exporter_version }}/sha256sums.txt"
+      mysqld_exporter_web_listen_address:
+        description: "Address on which mysqld_exporter will listen"
+        default: "0.0.0.0:9104"
+      mysqld_exporter_web_telemetry_path:
+        description: "Path under which to expose metrics"
+        default: "/metrics"
+      mysqld_exporter_enabled_collectors:
+        description:
+          - "List of dicts defining additionally enabled collectors and their configuration."
+          - "It adds collectors to L(those enabled by default,https://github.com/prometheus/mysqld_exporter#enabled-by-default)."
+        type: "list"
+        default: []
+      mysqld_exporter_disabled_collectors:
+        description:
+          - "List of disabled collectors."
+          - "By default mysqld_exporter disables collectors listed L(here,https://github.com/prometheus/mysqld_exporter#disabled-by-default)."
+        type: "list"
+        elements: "str"
+      mysqld_exporter_tls_server_config:
+        description:
+          - "Configuration for TLS authentication."
+          - "Keys and values are the same as in L(mysqld_exporter docs,https://github.com/prometheus/mysqld_exporter/blob/master/https/README.md#sample-config)."
+        type: "dict"
+      mysqld_exporter_http_server_config:
+        description:
+          - "Config for HTTP/2 support."
+          - "Keys and values are the same as in L(mysqld_exporter docs,https://github.com/prometheus/mysqld_exporter/blob/master/https/README.md#sample-config)."
+        type: "dict"
+      mysqld_exporter_basic_auth_users:
+        description: "Dictionary of users and password for basic authentication. Passwords are automatically hashed with bcrypt."
+        type: "dict"
```

Comment:
There are a few variables defined in `defaults/main.yml` that are missing here.

Some of them are also missig from the readme, but we can perhaps skip updating the readme because it will soon be obsolete in favor of docs that are auto generated from the argument specs.